### PR TITLE
fix: make wbtc the quote denom in upgrade handler

### PR DESCRIPTION
**Pull request by czarcas7ic**
Closes: #XXX

## What is the purpose of the change

Johnny pointed out that the WBTC pool in the upgrade handler had OSMO as the quote asset, when WBTC should supercede OSMO as the quote asset.

[I made an issue](https://github.com/osmosis-labs/osmosis/issues/6052) to automate this ranking, Sunny says the order should be:
USDC
USDT
DAI
WBTC
ETH
ATOM
OSMO

For now, this unblocks upgrade.

## Testing and Verifying

Go test changed, but we need to do this off a state exported testnet to ensure it works @alessandrolomanto @niccoloraspa 

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

File: app/upgrades/v17/constants.go [hunk 1]
```diff
@@ -395,9 +439,17 @@ func InitializeAssetPairsTestnet(ctx sdk.Context, keepers *keepers.AppKeepers) (
...
-	respectiveBaseAsset, err := keepers.GAMMKeeper.CalcOutAmtGivenIn(ctx, cfmmPool, sdk.NewCoin(QuoteAsset, sdk.NewInt(100000)), baseAsset, spreadFactor)
+	var respectiveBaseAsset sdk.Coin
+	var err error
+	if baseAsset == OSMO {
+		respectiveBaseAsset, err = keepers.GAMMKeeper.CalcOutAmtGivenIn(ctx, cfmmPool, sdk.NewCoin(baseAsset, sdk.NewInt(100000)), quoteAsset, spreadFactor)
+	} else if quoteAsset == OSMO {
+		respectiveBaseAsset, err = keepers.GAMMKeeper.CalcOutAmtGivenIn(ctx, cfmmPool, sdk.NewCoin(quoteAsset, sdk.NewInt(100000)), baseAsset, spreadFactor)
+	} else {
+		return fmt.Errorf("expected one of the pool's assets to be OSMO, got %s and %s", baseAsset, quoteAsset)
+	}
```
Comment (czarcas7ic) [inline app/upgrades/v17/constants.go:452]: This is only used for testnet, so not much scrutiny needed here

File: app/upgrades/v17/constants.go [hunk 2]
```diff
@@ -84,15 +88,28 @@ func InitializeAssetPairs(ctx sdk.Context, keepers *keepers.AppKeepers) ([]Asset
...
+				found = true
 				break
 			}
 		}
 
+		// If the quote asset is the only asset in the pool, return an error.
+		if !found {
+			return nil, errors.New("quote asset is the only asset in the pool")
+		}
+
```
Comment (ValarDragon) [inline app/upgrades/v17/constants.go:112]: LGTM

File: app/upgrades/v17/constants.go [hunk 3]
```diff
@@ -146,156 +163,182 @@ var (
 
 var AssetPairsForTestsOnly = []AssetPair{
 	{
+		QuoteAsset:        OSMO,
```
Comment (ValarDragon) [inline app/upgrades/v17/constants.go:166]: lol at having every asset in our test list

File: app/upgrades/v17/upgrades.go [hunk 4]
```diff
@@ -138,22 +138,30 @@ func CreateUpgradeHandler(
 }
 
 // createCLPoolWithCommunityPoolPosition creates a CL pool for a given balancer pool and adds a full range position with the community pool.
-func createCLPoolWithCommunityPoolPosition(ctx sdk.Context, keepers *keepers.AppKeepers, gammPoolId uint64, baseAsset string, spreadFactor sdk.Dec, communityPoolAddress sdk.AccAddress) (clPoolDenom string, clPoolId uint64, poolLink gammmigration.BalancerToConcentratedPoolLink, coinsUsed sdk.Coins, err error) {
+func createCLPoolWithCommunityPoolPosition(ctx sdk.Context, keepers *keepers.AppKeepers, assetPair AssetPair, communityPoolAddress sdk.AccAddress) (clPoolDenom string, clPoolId uint64, poolLink gammmigration.BalancerToConcentratedPoolLink, coinsUsed sdk.Coins, err error) {
```
Comment (ValarDragon) [inline app/upgrades/v17/upgrades.go:141]: Note to self, make a devbot better-diff command

Comment (ValarDragon) [reply app/upgrades/v17/upgrades.go:141]: using AssetPair is a good decision

Comment (ValarDragon) [reply app/upgrades/v17/upgrades.go:141]: Function changes make sense!

File: app/upgrades/v17/upgrades_test.go [hunk 5]
```diff
@@ -85,29 +84,19 @@ func assertEqual(suite *UpgradeTestSuite, pre, post interface{}) {
 }
 
 func (suite *UpgradeTestSuite) TestUpgrade() {
-	upgradeSetup := func() {
-		// This is done to ensure that we run the InitGenesis() logic for the new modules
-		upgradeStoreKey := suite.App.AppKeepers.GetKey(upgradetypes.StoreKey)
-		store := suite.Ctx.KVStore(upgradeStoreKey)
-		versionStore := prefix.NewStore(store, []byte{upgradetypes.VersionMapByte})
-		versionStore.Delete([]byte(cltypes.ModuleName))
-	}
```
Comment (ValarDragon) [inline app/upgrades/v17/upgrades_test.go:94]: This code was just a bug lol

**Review by ValarDragon — APPROVED** [review]
LGTM, nice job

**Review by AlpinYukseloglu — APPROVED** [review]
LGTM, i like the asset pair change. Note that my review of the upgrade tests was cursory

**Review by p0mvn — APPROVED** [review]
LGTM

**Comment by czarcas7ic** [comment]
Upgrade handler confirmed to work with state exported testnet, merging

**Pull request merged**